### PR TITLE
week 1 assignment completed

### PR DESCRIPTION
**Pull request by Himanshi9gupta**

### CODE REVIEW

File: dataInterface/items.js [hunk 1]
```diff
@@ -2,22 +2,26 @@ const uuid = require('uuid');
...
+    return module.exports.items.splice(index, 1);
 }
 
 module.exports.updateById = async (itemId, newObj) => {
-    // TODO: complete writing this function
+   const element = module.exports.items.find(ele => ele.id === itemId)
+   if(element){
+    element.field = newObj.field;
+   }
 }
```
Comment (cherimarie) [inline dataInterface/items.js:25]: If the return value of this function is returned to the caller from the route handler, shouldn't this return something?

File: routes/items.js [hunk 2]
```diff
@@ -18,14 +22,17 @@ router.post("/", (req, res, next) => {
...
-
 router.delete("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+ if(itemData.getById(req.params.id)){
+  res.status(200).send(itemData.deleteById(req.params.id));
+ }
+  else {
+    res.status(404).json({error: "item does not exist"})
+  }
```
Comment (cherimarie) [inline routes/items.js:35]: The logic of whether the item exists should be handled by the data interface method. The logic of what to do based on whether the item is found is great. Maybe `deleteById` could return either the deleted item or null, depending on whether it's found?

File: routes/items.test.js [hunk 3]
```diff
@@ -81,6 +81,10 @@ describe("/items", () => {
       expect(itemData.items.length).toEqual(1);
       expect(itemData.items).toEqual([items[0]]);
     });
+    it("should return 404 if item does not exist", async () => {
+      const res = await request(server).get("/items/other");
+      expect(res.statusCode).toEqual(404);
+    });
```
Comment (cherimarie) [inline routes/items.test.js:87]: This isn't a new test. If we want to test that the delete route handler returns a 404 as appropriate, we need to test that path.
```
it("should return 404 if item does not exist", async () => {
      const res = await request(server).delete("/items/other");
      expect(res.statusCode).toEqual(404);
    });
```

File: dataInterface/items.js (hunk 1 above)
Comment (Himanshi9gupta) [reply dataInterface/items.js:25]: yes, it should return the element object.

File: routes/items.js (hunk 2 above)
Comment (Himanshi9gupta) [reply routes/items.js:35]: Cheri, I am bit confused here how it makes difference if we handle item found condition in the data interface method or in the router method handler ? is it about best practices ?

File: routes/items.test.js (hunk 3 above)
Comment (Himanshi9gupta) [reply routes/items.test.js:87]: path you mean URL ?

**Pull request closed without merging**